zeroex/orderwatch/order_watcher.go

```diff
@@ -1157,42 +1159,46 @@ func (w *Watcher) convertValidationResultsIntoOrderEvents(
 			}
 			orderEvents = append(orderEvents, orderEvent)
 		} else {
-			expiration := time.Unix(order.SignedOrder().ExpirationTimeSeconds.Int64(), 0)
+			// The order is expired if ExpirationTimeSeconds is equal to or less than the timestamp
+			// of the validation block.
+			validationBlockTimeStampSeconds := big.NewInt(validationBlock.Timestamp.Unix())
+			isOrderExpired := order.ExpirationTimeSeconds.Cmp(validationBlockTimeStampSeconds) != 1
 
 			if oldFillableAmount.Cmp(newFillableAmount) == 0 {
 				// If order was previously expired, check if it has become unexpired
-				if order.IsRemoved && oldFillableAmount.Cmp(big.NewInt(0)) != 0 && validationBlockTimestamp.Before(expiration) {
-					w.rewatchOrder(order, nil)
+				if order.IsRemoved && oldFillableAmount.Cmp(big.NewInt(0)) != 0 && !isOrderExpired {
```

Comment:
It would be nice to have `isOrderUnexpired := order.IsRemoved && oldFillableAmount.Cmp(big.NewInt(0)) != 0 && !isOrderExpired` and use it in the threelocations where we check.

Reply:
Good suggestion. I know this logic is a bit of a mess right now. It's something we should look at again when we have time to address tech debt. This PR is just trying to do the bare minimum.